<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideRight</enum>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>红黑树</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>GroupBox</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QSpinBox" name="brAddSpinBox">
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>0</height>
              </size>
             </property>
             <property name="minimum">
              <number>-1000</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="brAddBtn">
             <property name="text">
              <string>添加</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="brDelSpinBox">
             <property name="minimumSize">
              <size>
               <width>75</width>
               <height>0</height>
              </size>
             </property>
             <property name="minimum">
              <number>-1000</number>
             </property>
             <property name="maximum">
              <number>1000</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="brDelBtn">
             <property name="text">
              <string>删除</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="clearBtn">
             <property name="text">
              <string>清除</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="TreePainter" name="treePainter" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>寻路算法</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="topMargin">
           <number>12</number>
          </property>
          <property name="bottomMargin">
           <number>12</number>
          </property>
          <item>
           <widget class="MazePainter" name="mazePainter" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>1</horstretch>
              <verstretch>1</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="title">
           <string>GroupBox</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QComboBox" name="option">
             <item>
              <property name="text">
               <string>A*最优寻路</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>dfs深度优先</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>bfs广度优先</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>抽象模式</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="findBtn">
             <property name="text">
              <string>开始寻路</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="stopBtn">
             <property name="text">
              <string>停止寻路</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="resetBtn">
             <property name="text">
              <string>重置</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TreePainter</class>
   <extends>QWidget</extends>
   <header>treepainter.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>MazePainter</class>
   <extends>QWidget</extends>
   <header>mazepainter.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>brDelSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>brDelBtn</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>247</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>brAddSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>brAddBtn</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>87</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>167</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
